src/ximea_ros_cam.cpp

```diff
@@ -328,18 +332,24 @@ void XimeaROSCam::initCam() {
 void XimeaROSCam::openCam() {
     // Init variables
     XI_RETURN xi_stat;
-
+    int cam_detect_poll_count = 0;
 
     // Disable auto bandwidth calculation (before camera open)
     // Do this if bandwidth limitation is required or desired
     if (this->cam_num_in_bus_ > 1) {
         xiSetParamInt(0, XI_PRM_AUTO_BANDWIDTH_CALCULATION, XI_OFF);
     }
 
-    // Open camera
-    xi_stat = xiOpenDeviceBy(XI_OPEN_BY_SN,
-                             this->cam_serialno_.c_str(),
-                             &this->xi_h_);
+    xi_stat = XI_NO_DEVICES_FOUND;
+
+    // Loop until camera is detected on the bus or max poll detection exceeded
+    while(xi_stat == XI_NO_DEVICES_FOUND &&
+          cam_detect_poll_count < this->cam_detect_max_polls_) {
+      xi_stat = xiOpenDeviceBy(XI_OPEN_BY_SN,
+                               this->cam_serialno_.c_str(),
+                               &this->xi_h_);
+      cam_detect_poll_count++;
```

Comment:
Can we have an option for an indefinite poll amount? Such as it will attempt to connect to the camera every 2 to 3 seconds until we kill the roslaunch via a Ctrl-C or another method.